new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsShutdownOption.java

```diff
@@ -183,108 +288,117 @@ private static void createAndVerifyMiiDomain() {
+                    .shutdown(shutDownObject[4]))));
+    return domain;
+  }
 
+  // create domain resource and verify all the server pods are ready
+  private void createVerifyDomain(Domain domain) {
     // create model in image domain
     logger.info("Creating model in image domain {0} in namespace {1} using docker image {2}",
         domainUid, domainNamespace, miiImage);
     createDomainAndVerify(domain, domainNamespace);
 
-    // check that admin server pod exists in the domain namespace
-    logger.info("Checking that admin server pod {0} exists in namespace {1}",
+    // check that admin service exists in the domain namespace
+    logger.info("Checking that admin service {0} exists in namespace {1}",
         adminServerPodName, domainNamespace);
-    checkPodExists(adminServerPodName, domainUid, domainNamespace);
+    checkServiceExists(adminServerPodName, domainNamespace);
 
     // check that admin server pod is ready
     logger.info("Checking that admin server pod {0} is ready in namespace {1}",
         adminServerPodName, domainNamespace);
     checkPodReady(adminServerPodName, domainUid, domainNamespace);
 
-    // check that admin service exists in the domain namespace
-    logger.info("Checking that admin service {0} exists in namespace {1}",
-        adminServerPodName, domainNamespace);
-    checkServiceExists(adminServerPodName, domainNamespace);
-
-    // check for managed server pods existence in the domain namespace
     for (int i = 1; i <= replicaCount; i++) {
-      String managedServerPodName = managedServerPrefix + i;
+      String managedServerPodName = managedServerPodNamePrefix + i;
 
-      // check that the managed server pod exists in the domain namespace
-      logger.info("Checking that managed server pod {0} exists in namespace {1}",
+      // check that the managed server service exists in the domain namespace
+      logger.info("Checking that managed server service {0} exists in namespace {1}",
           managedServerPodName, domainNamespace);
-      checkPodExists(managedServerPodName, domainUid, domainNamespace);
+      checkServiceExists(managedServerPodName, domainNamespace);
 
       // check that the managed server pod is ready
       logger.info("Checking that managed server pod {0} is ready in namespace {1}",
           managedServerPodName, domainNamespace);
       checkPodReady(managedServerPodName, domainUid, domainNamespace);
+    }
 
+    // check for independent managed server pods existence in the domain namespace
+    for (String podName : new String[]{indManagedServerPodName2}) {
       // check that the managed server service exists in the domain namespace
       logger.info("Checking that managed server service {0} exists in namespace {1}",
-          managedServerPodName, domainNamespace);
-      checkServiceExists(managedServerPodName, domainNamespace);
+          podName, domainNamespace);
+      checkServiceExists(podName, domainNamespace);
```

Comment:
you can call checkPodReadyAndServiceExists(String podName, String domainUid, String namespace) method

Reply:
modified